### PR TITLE
[BE]: Update cudnn to 8.9.7.29

### PR DESCRIPTION
**Pull request by Skylion007**
Update cudnn to 8.9.7.29 . We just updated the cudnn frontend, might as well. Mostly has improvements for the cudnn flash attention implementation which we are interested in exploring. Such as in #115663

### CODE REVIEW

**Review by eqy — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Comment by eqy** [comment]
Looks like there's an inductor failure potentially stemming from using the default threshold where cuDNN could potentially use TF32
https://github.com/pytorch/pytorch/blob/237773132d7f9ee474aff444906c133c49cb253b/test/inductor/test_layout_optim.py#L234

**Comment by Skylion007** [comment]
What's should we do to fix the failing case @eqy ?

**Comment by eqy** [comment]
@Skylion007 let me see if I can reproduce the failure on my local setup on sm86, which should have cuDNN 8.9.7+---I'll submit a PR to hopefully fix it

**Comment by Skylion007** [comment]
@eqy seems like that didn't fix it. :(

**Comment by malfet** [comment]
@Skylion007 PyTorch binary build system is split between this repo and builder and for lots of bizzare reason one needs to duplicate the logic across the repo. Let me push a change that will update cudnn version on the builder side

**Review by huydhn — APPROVED** [review]

**Comment by eqy** [comment]
Let me ping cuDNN about it as 8.9.7 is among the more well-tested released and should be hosted at this point

**Comment by ptrblck** [comment]
Yes, @malfet is correct and `nvidia-cudnn-cu11==8.9.7.29` is not available as cuDNN is running into the project size limitations.
RFE to increase the size limit for `nvidia-cudnn-cu11`: https://github.com/pypi/support/issues/3402
RFE to increase the size limit for `nvidia-cudnn-cu12`: https://github.com/pypi/support/issues/3408

Both were opened in ~Dec. 2023 and we haven't received any updates yet. Unsure if @rgommers would know more

**Comment by Skylion007** [comment]
It would be nice to update to the latest CUDNN that we can support for 2.3. What's the latest version that is feasible to support?

**Comment by malfet** [comment]
I can update our index to 8.9.7.29 from NVIDIA's pypi that should unblock both this change and the release

File: .ci/docker/common/install_cudnn.sh [hunk 1]
```diff
@@ -5,11 +5,11 @@ if [[ ${CUDNN_VERSION} == 8 ]]; then
     mkdir tmp_cudnn
     pushd tmp_cudnn
     if [[ ${CUDA_VERSION:0:4} == "12.1" ]]; then
-        CUDNN_NAME="cudnn-linux-x86_64-8.9.2.26_cuda12-archive"
+        CUDNN_NAME="cudnn-linux-x86_64-8.9.7.29_cuda12-archive"
```
Comment (atalman) [inline .ci/docker/common/install_cudnn.sh:8]: Please update only 11.8 cudnn version. At this point we want to upgrade only cuda 11.8 cudnn.

Comment (atalman) [reply .ci/docker/common/install_cudnn.sh:8]: @ptrblck Sorry yes, issue exist in cudnn 11, my bad https://pypi.org/project/nvidia-cudnn-cu12/#files. I do see  ``cudnn 8.9.7.29`` available.

**Comment by atalman** [comment]
@Skylion007 @malfet Here are additional work required for cudnn : https://github.com/pytorch/builder/blob/main/CUDA_UPGRADE_GUIDE.MD#upgrade-cudnn-version-only
For Windows we would need to rebuild the AMI.

**Comment by Skylion007** [comment]
Closing in favor of #123475

**Comment by nWEIdia** [comment]
Closing as v9 is in :)

**Pull request closed without merging**